### PR TITLE
Use derive_more in more places

### LINKED ISSUE #8: Check where derive_more may be used instead of a raw trait implementation
This mostly affect `Display` and `Debug` traits.

### PR DESCRIPTION
**Pull request by i-schuetz**
Closes #8

### CODE REVIEW

File: algonaut_core/src/lib.rs [hunk 1]
```diff
@@ -296,19 +267,6 @@ impl Debug for Signature {
     }
 }
 
-impl PartialEq for Signature {
```
Comment (i-schuetz) [inline algonaut_core/src/lib.rs:299]: Seems to not be necessary.

**Comment by manuelmauro** [comment]
Would it be possible to have some tests and some references to `derive_more` docs to be sure that the new implementations are equivalent to the previous ones? For instance `Add` and `Sub` implementations are so standard that a simple reference to `derive_more` docs would convince us of their equivalence, while I am not completely sure of `PartialEq` for `Signature` for which some tests would be helpful.

**Comment by i-schuetz** [comment]
Hmm I wrote a test locally, to confirm, but was hesitant about uploading it, as it feels a bit like testing derive_more or the language. At least for simple things, I'd tend to rely on the dependencies, as otherwise we'd have to maintain too many tests.  Does that make sense?

**Comment by i-schuetz** [comment]
It's actually a bit difficult to reason about as we shouldn't make assumptions about how `PartialEq` is implemented, making it necessary to test it, but on the other side we're assuming that it's relatively complicated (compared to e.g. `Add`). 

I'll add it. In case of doubt, it's always better to have more tests.

File: algonaut_core/src/lib.rs [hunk 2]
```diff
@@ -401,4 +359,38 @@ mod tests {
 
         assert!(invalid_csum.parse::<Address>().is_err());
     }
+
+    /// PartialEq is implemented correctly (signatures are equal)
+    #[test]
+    fn signatures_equal() {
```
Comment (i-schuetz) [inline algonaut_core/src/lib.rs:365]: Possible shorter version: `assert_eq!(Signature([0; 64]), Signature([0; 64]))`. But explicit (and not repeated) is better.

**Comment by manuelmauro** [comment]
Thanks, if you tested the equivalence locally that's enough for me! We can avoid keeping the tests as you suggested.

**Pull request merged**